### ISSUE TITLE
Warning in readShapes()

### ISSUE THREAD

**Issue by miltontan** [aaronolsen/StereoMorph#8]
Hi, so I'm getting a warning in R 4.0.2 using readShapes() 

In if (class(read_shapes[[i]][[name]]) %in% c("character", ...:
   the condition has length > 1 and only the first element will be used
In if (class(read_shapes[[i]][[name]]) %in% c("matrix", ...:
   the condition has length > 1 and only the first element will be used

I presume this is because matrix objects now have two classes in R 4: c("matrix", "array"), if I'm not mistaken?

**Comment by aaronolsen** [comment]
Thanks for posting this Milton! I just tried readShapes() on an old shapes file on my system using R 4.0.2 and I didn't get a warning message. Could you send me the shapes file that's giving the warning so I can try it on my system? You can attach it to another message or send it to me by e-mail, whichever you prefer. Not sure what structure in particular might be causing this.

**Comment by miltontan** [comment]
Full disclosure I saw the problem on my student's machine so I don't have all of the details (I'm still running R 3.6.0). But, what I did see is it didn't even work on the very first folder. Attached is not the same folder as the one we tested, but is theoretically the same structure. 

Note I was able to circumvent the issue by wrapping the conditional test in the if statements with any() to allow for testing if class() results in a vector, and I think it worked...

Thanks,

Milton

**Comment by miltontan** [comment]
As an addendum, the same warnings come up with reconstructStereoSets and calibrateCameras because they both call readShapes. Replacing with my modified version of readShapes fixes this issue. In addition, list2XML4R has a similar if statement testing for the class of objects being a matrix or array and cause additional warnings, which can be fixed similarly using any().

**Comment by aaronolsen** [comment]
Thanks so much @miltontan ! You've done all the debugging for me! :-) Since you've already found a fix and it's just a warning message I haven't moved on this immediately. But I will get to it soon!

**Comment by Ax3man** [comment]
Hey, also just ran into this when using `digitizeImages`. The recommended way to check for the class of an object is to use `inherits` or `is`, which avoid these kind of bugs going forward. See this helpful blog from Martin Maechler: https://developer.r-project.org/Blog/public/2019/11/09/when-you-think-class.-think-again/index.html. 

So, for example: https://github.com/aaronolsen/StereoMorph/blob/60a35da93b91c945faa4fae27d323308f5b6a6bc/R/list2XML4R.R#L25 can be rewritten as:
```
if (inherits(list[[name]], c('logical', 'character', 'integer', 'vector', 'numeric')))
```

**Comment by miltontan** [comment]
Hi, I have run into a new issue related to digitizeImages and related to these functions. When I load the updated list2XML4R and readShapes files so that those functions are in the environment, if I then load digitizeImages and save a landmark file, it inserts this extra line into the landmarks right before </landmarks.pixel> and </landmarks.scaled> like the below:

	<landmarks.pixel type=array dim=24,2 as.numeric=TRUE >

The digitizer app does not appear to expect this row, so if you either reopen the digitizer or simply page back and forth between the images, when the digitizer comes back to a file containing this extra line it displays all of the digitized landmarks as NaN. Saving again then replaces all of the landmark coordinates with NaN. 

I have confirmed that this line does not seem to appear if I have not loaded the modified versions of the functions, but of course then I get the warnings I described previously.
